File: app/src/main/java/com/shetj/diyalbume/ppttest/NewIndexActivity.java
package com.shetj.diyalbume.ppttest;

import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.shetj.diyalbume.R;

import java.util.ArrayList;
import java.util.List;

import me.shetj.base.base.BaseActivity;
import me.shetj.base.base.BaseMessage;
import me.shetj.base.tools.app.ArmsUtils;

/**
 * @author shetj
 */
public class NewIndexActivity extends BaseActivity<IndexPresenter> {

	private RecyclerView mIRecyclerView;
	private IndexAdpter indexAdpter ;
	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.activity_new_index);
		ArmsUtils.statuInScreen(this,true);
		initView();
		initData();
	}

	@Override
	protected void initView() {
		mIRecyclerView = findViewById(R.id.IRecyclerView);
		ArmsUtils.configRecycleView(mIRecyclerView,new LinearLayoutManager(this));
	}

	@Override
	protected void initData() {
		mPresenter = new IndexPresenter(this);
		indexAdpter = new IndexAdpter(new ArrayList<>());
		mIRecyclerView.setAdapter(indexAdpter);

		mPresenter.initData();
		addHeadView();
	}

	private void addHeadView() {

	}


	@Override
	public void updateView(@NonNull BaseMessage message) {
		super.updateView(message);
		switch (message.type){
			case 1:
				indexAdpter.setNewData((List<ItemIndex>) message.obj);
				break;
			case 2:
				break;
			default:
				break;

		}
	}
}

File: app/src/main/java/com/shetj/diyalbume/ppttest/NewIndexActivity.kt
package com.shetj.diyalbume.ppttest

import android.os.Bundle
import android.os.Message
import androidx.recyclerview.widget.RecyclerView
import com.shetj.diyalbume.R
import me.shetj.base.base.BaseActivity
import me.shetj.base.tools.app.ArmsUtils.Companion.statuInScreen
import java.util.*

/**
 * @author shetj
 */
class NewIndexActivity : BaseActivity<IndexPresenter>() {

    private var mIRecyclerView: RecyclerView? = null
    private var indexAdpter: IndexAdpter? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_new_index)
        statuInScreen( true)
    }

    override fun initView() {
        mIRecyclerView = findViewById(R.id.IRecyclerView)
        mIRecyclerView?.scrollX = 1
    }

    override fun initData() {
        mPresenter = IndexPresenter(this)
        indexAdpter = IndexAdpter(ArrayList())
        mIRecyclerView!!.adapter = indexAdpter

        mPresenter!!.initData()
        addHeadView()
    }

    private fun addHeadView() {

    }


    override fun updateView(message: Message) {
        super.updateView(message)
        when (message.what) {
            1 -> indexAdpter!!.setNewData(message.obj as MutableList<ItemIndex>)
            2 -> {
            }
            else -> {
            }
        }
    }
}
